### ISSUE TITLE
ReportGenerator 4.3.0 breaks the build

### ISSUE THREAD

**Issue by david-driscoll** [nuke-build/nuke#318]
It appears the latest [`ReportGenerator`](https://www.nuget.org/packages/ReportGenerator/4.3.0) is doing some additional multi-targeting (adding `netcoreapp3.0` to it's list).  This appears to be making the `ToolPathResolver` fail.  It doesn't fail today when the previous version because there was only executable in the package, now there are two.

I tried to see if I could find an example of selecting a framework to update the specification but that remained unclear to me.

```
Assertion failed: Package executable ReportGenerator.exe [ReportGenerator] requires a framework: net47, netcoreapp3.0
   at Nuke.Common.ControlFlow.Fail(String text)
   at Nuke.Common.Tooling.ToolPathResolver.GetPackageExecutable(String packageId, String packageExecutable, String version, String framework)
   at Nuke.Common.Tooling.ProcessTasks.StartProcess(ToolSettings toolSettings)
   at Nuke.Common.Tools.ReportGenerator.ReportGeneratorTasks.ReportGenerator(ReportGeneratorSettings toolSettings)
   at Rocket.Surgery.Nuke.RocketBoosterBuild.<get_Generate_Code_Coverage_Reports>b__33_2() in /_/src/Nuke/RocketBoosterBuild.cs:line 184
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, ExecutableTarget target, IReadOnlyCollection`1 previouslyExecutedTargets, Boolean failureMode)
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, IReadOnlyCollection`1 skippedTargets)
   at Nuke.Common.Execution.BuildManager.Execute[T](Expression`1[] defaultTargetExpressions)
```

**Comment by matkoch** [comment]
`SetFramework` :)

**Comment by arodus** [comment]
netcore3.0 allows building of executables again, because of that there are now multiple ReportGenerator.exe available. I'm pretty sure this will happen more frequently in the future so we probably have to find a way to handle that in Nuke directly (@matkoch).


There is no `SetFramework` available for ReportGenerator but should be similar as for xunit.

**Comment by david-driscoll** [comment]
Indeed if there was a way to set the framework that is used (if there was a work around I would do that as well)

It fails here https://github.com/nuke-build/common/blob/6f9d87f6114433122f8ea131b92022e3c8d5f6dc/source/Nuke.Common/Tools/ReportGenerator/ReportGenerator.Generated.cs#L33-L35

It should also be possible to infer the framework that will work by looking at the currently executing runtime and make the most logical choice, instead of just dismissing all the choices.

**Comment by matkoch** [comment]
@david-driscoll seems like I've confused it. It's not generated, but manually implemented (for instance for xunit), and it's available from the `ToolPathResolver`.. I will take care in a bit.

**Comment by arodus** [comment]
@david-driscoll you can set the `ToolPath` manually as a workaround:
```csharp
ReportGeneratorTasks.ReportGenerator(s => s
    .SetToolPath(ToolPathResolver.GetPackageExecutable("ReportGenerator", "ReportGenerator.exe",
                    "netcoreapp3.0")));

```

**Comment by david-driscoll** [comment]
@arodus thanks for the tip! Right now I'm not rush to update I just figured the problem would show up here eventually! 👍

**Comment by Rookian** [comment]
> @david-driscoll you can set the `ToolPath` manually as a workaround:
> 
> ```cs
> ReportGeneratorTasks.ReportGenerator(s => s
>     .SetToolPath(ToolPathResolver.GetPackageExecutable("ReportGenerator", "ReportGenerator.exe",
>                     "netcoreapp3.0")));
> ```

@arodus You mixed up the given parameters.
The correct call should be:
 ```cs
 ReportGeneratorTasks.ReportGenerator(s => s
     .SetToolPath(ToolPathResolver.GetPackageExecutable("ReportGenerator", "ReportGenerator.exe",
                     "4.3.6", "netcoreapp3.0")));
 ```
This is quite ugly, but it works.

Does Nuke support the new "local tools" option in Dotnet Core 3.0?

**Comment by matkoch** [comment]
@Rookian time for a new issue. The given one is resolved with 0.23.0. And please also star the project 😉
